### ISSUE TITLE
All source code is cleared if html syntax error entered

### ISSUE THREAD

**Issue by Tanatole** [ckeditor/ckeditor4#5029]
## Type of report

Bug

## Provide detailed reproduction steps (if any)
1. Enter a text, example "Hello".
2. Switch to source view. Source is **<p>Hello</p>**
3. Change with <p **class="title"**>Hello</p>
4. Switch back to preview and again to source view.
5. Code has been parsed, class is removed. Source code is now <p>Hello</p>
6. Change it again with syntax error : <p **class="title**>Hello</p> (forgot last " character)
7. Switch to preview. All is blank. You've just lost all your code !!!

### Expected result

At least, no change to source when syntax error encountered.

It would be great to be able to not parse and "clean" source code, because in some cases we **need** to add some css classes or ids for instance.

### Actual result

Source code is lost.

## Other details

Tested with ckeditor 4.4.0 and 4.12.1 (php / javascript environment)

**Comment by Tanatole** [comment]
By the way, it's maybe already possible to disable this parsing in ckeditor config file, but I don't get how to do that. Sorry...

**Comment by Comandeer** [comment]
It's more of a feature request to add handling of syntax errors in input data. I've changed the title and type of issue accordingly.

**Comment by Augustin82** [comment]
> It's more of a feature request to add handling of syntax errors in input data. I've changed the title and type of issue accordingly.

Not really, because in the original description it says that valid HTML attributes are removed. So it really is about "disabling parsing/rewriting" of code, isn't it?
